Core/Core/Planner/CalendarEventDetails/ViewModel/CalendarEventDetailsViewModel.swift

```diff
@@ -0,0 +1,109 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import SwiftUI
+
+public class CalendarEventDetailsViewModel: ObservableObject {
+    // MARK: Output
+    @Published public private(set) var state: InstUI.ScreenState = .loading
+    @Published public private(set) var title: String?
+    @Published public private(set) var date: String?
+    @Published public private(set) var locationInfo: [InstUI.TextSectionView.SectionData] = []
+    @Published public private(set) var details: InstUI.TextSectionView.SectionData?
```

Comment:
I know these definitions make things easier on the view side, I'm just not sure if we want to introduce such coupling between the "soon to be Design System" and the view model.

Reply:
I think the benefits are bigger than the drawbacks at this time although we can talk about how to approach this in the future.